crates/pallet-domains/src/lib.rs

```diff
@@ -932,7 +931,43 @@ impl<T: Config> Pallet<T> {
 
         // TODO: Implement bundle validation.
 
-        // TODO: Verify ProofOfElection
+        // TODO: The current staking distribution may be unusable when there is an epoch
+        // transition, track the last stake distribution in that case.
+        let proof_of_election = &sealed_header.header.proof_of_election;
+
+        proof_of_election
+            .verify_vrf_signature(&operator.signing_key)
+            .map_err(|_| BundleError::BadVrfSignature)?;
+
+        let domain_id = proof_of_election.domain_id;
+
+        let domain_stake_summary =
+            DomainStakingSummary::<T>::get(domain_id).ok_or(BundleError::InvalidDomainId)?;
+
+        if !domain_stake_summary
+            .current_operators
```

Comment:
I see we can actually make this (operator_id, their_stake), then this would be much simpler with what were discussing with managing staking history when epoch change happens when new head is added instead of `on_finalize`. Non blocker thgouh

Reply:
Yeah, that would make it simpler to track the last epoch's staking distribution, and no extra iteration is required. We can update it once the epoch transition lands.